### PR TITLE
Add proprietary_codecs.gn

### PR DESCRIPTION
**Pull request by niranjanyardi**
b/377271824

### CODE REVIEW

File: cobalt/build/configs/media.gn [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Build flag declared originally in build/config/features.gni
+# This flag is overridden to enable common video codecs playbacks.
+proprietary_codecs = true
+
+ffmpeg_branding = "Chrome"
```
Comment (niranjanyardi) [inline cobalt/build/configs/media.gn:5]: @borongc any suggestions as to why this is needed. Can you suggest a 1 line why this is needed?

Gets declared in chromecast/build/args/cast_shell_android.gn

Comment (andrewsavage1) [reply cobalt/build/configs/media.gn:5]: See https://source.chromium.org/chromium/chromium/src/+/main:android_webview/docs/aosp-system-integration.md;l=274?q=ffmpeg_branding%20f:%5C.md&ss=chromium%2Fchromium%2Fsrc

**Review by yell0wd0g — APPROVED** [review]
FWIW this change makes sense to me.

Has the added value of the ease of adding specific media-flags in the future.

File: cobalt/build/configs/linux-x64x11/args.gn [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import("cobalt/build/configs/media.gn")
```
Comment (andrewsavage1) [inline cobalt/build/configs/linux-x64x11/args.gn:1]: This should be removed, and I think `media.gn` could be better named (i.e. to proprietary_codecs.gn)

Comment (niranjanyardi) [reply cobalt/build/configs/linux-x64x11/args.gn:1]: Agree with the rename ,  the import is still needed, correct? :

import("cobalt/build/configs/proprietary_codecs.gn")

File: cobalt/build/configs/media.gn (hunk 1 above)
Comment (niranjanyardi) [reply cobalt/build/configs/media.gn]: Thanks for the links, added them to docs!

**Review by andrewsavage1 — APPROVED** [review]

**Review by borongc — APPROVED** [review]

**Review by oxve — APPROVED** [review]

**Pull request merged**